### ISSUE TITLE
Use py.test?

### ISSUE THREAD

**Issue by tbekolay** [nengo/nengo#196]
I know I was the one that pushed for using the standard library's unittest instead of nose, but I'm just finishing some work on direct mode, and thinking about how to get more unittests using multiple nonlinearity types feels like the unittest solution is going to be cumbersome. Having nicer syntax for tests should make them less of a burden to write, and py.test lets us run tests in parallel (helpful, at least, for the reference implementation, not quite as helpful for nengo_ocl). More and more projects that I run across use py.test, and it works on Python 2 and 3, so it seems like a good solution for the foreseeable future.

Any arguments against using py.test? If not, I'll at least attempt to switch over (not now, probably sometime in December) and see if issues arise.
